server/aap/import_text_archive/commands.py

```diff
@@ -198,7 +206,46 @@ def _process_bunch(self, x):
                 generate_unique_id_and_name(item)
                 item['ingest_id'] = id
 
+                last_line = None
+                el = doc.find('dcdossier/document/body/BodyText')
+                if el is not None:
+                    story = el.text
+                    lines = story.split('\n')
+                    if len(lines) > 0:
+                        last_line = lines[-1]
+                    if item.get(FORMAT) == FORMATS.HTML:
+                        story = story.replace('\n   ', '<p></p>')
+                        story = story.replace('\n', '<br>')
+                        item['body_html'] = '<p>' + story + '</p>'
+                    else:
+                        item['body_html'] = '<pre>' + story + '</pre>'
+                    try:
+                        item['word_count'] = get_text_word_count(item['body_html'])
+                    except:
+                        pass
+
                 item['source'] = self._get_head_value(doc, 'Agency')
+                if item['source'] is None:
```

Comment:
It is a different kind of wrong to the other cases, this is if the agency is missing all together rather than being there but wrong!